src/pocketmine/plugin/PluginManager.php

```diff
@@ -362,39 +370,80 @@ public function loadPlugins(string $directory, array $newLoaders = null){
 	 * @return bool
 	 */
 	public function isCompatibleApi(string ...$versions) : bool{
-		$serverString = $this->server->getApiVersion();
-		$serverApi = array_pad(explode("-", $serverString), 2, "");
-		$serverNumbers = array_map("intval", explode(".", $serverApi[0]));
+		$required = $this->server->getApiVersion();
 
 		foreach($versions as $version){
-			//Format: majorVersion.minorVersion.patch (3.0.0)
-			//    or: majorVersion.minorVersion.patch-devBuild (3.0.0-alpha1)
-			if($version !== $serverString){
-				$pluginApi = array_pad(explode("-", $version), 2, ""); //0 = version, 1 = suffix (optional)
+			if(self::compareApi($required, $version)){
+				return true;
+			}
+		}
 
-				if(strtoupper($pluginApi[1]) !== strtoupper($serverApi[1])){ //Different release phase (alpha vs. beta) or phase build (alpha.1 vs alpha.2)
-					continue;
-				}
+		return false;
+	}
 
-				$pluginNumbers = array_map("intval", array_pad(explode(".", $pluginApi[0]), 3, "0")); //plugins might specify API like "3.0" or "3"
+	/**
+	 * Returns whether a given API version is compatible with the required version.
+	 *
+	 * @param string $haveVersion
+	 * @param string $needVersion
+	 *
+	 * @return bool
+	 */
+	public static function compareApi(string $haveVersion, string $needVersion) : bool{
```

Comment:
"Compare" is inaccurate. You aren't comparing versions here; you are checking if they are compatible. Consider renaming this function (and maybe `isCompatibleApi` too)

Reply:
`isCompatibleApi` can't be renamed here due to BC concerns.